include/tunnler/room_member.h

```diff
@@ -23,6 +24,14 @@
 // It also has to be used if you host a game yourself (You'd create both, a Room and a RoomMembership for yourself)
 class RoomMember final {
 public:
+    using Callback = std::shared_ptr<std::function<void()> >;
```

Comment:
I just felt it weird to have a typedef for callback functions in C++ (sounds like a C thing to me). Also considering std::function is pretty much made for that purpose. I'm also not sure how convenient a shared_ptr is for std::bind etc.

So I'd probably just follow what the std C++ lib does or commonly used libs / APIs.